mailbox/store/src/main/java/org/apache/james/mailbox/store/search/LazyMessageSearchIndex.java

```diff
@@ -1,150 +1,159 @@
-        this.factory = factory;
-    }
-
-    @Override
-    public Group getDefaultGroup() {
-        return GROUP;
-    }
-    
-    @Override
-    public EnumSet<SearchCapabilities> getSupportedCapabilities(EnumSet<MailboxManager.MessageCapabilities> messageCapabilities) {
-        return EnumSet.noneOf(SearchCapabilities.class);
-    }
-
-    @Override
-    public Mono<Void> add(MailboxSession session, Mailbox mailbox, MailboxMessage message) {
-        return index.add(session, mailbox, message);
-    }
-
-    @Override
-    public Mono<Void> delete(MailboxSession session, Mailbox mailbox, Collection<MessageUid> expungedUids) {
-        return index.delete(session, mailbox, expungedUids);
-    }
-
-    @Override
-    public Mono<Void> deleteAll(MailboxSession session, MailboxId mailboxId) {
-        return index.deleteAll(session, mailboxId);
-    }
-
-    /**
-     * Lazy index the mailbox on first search request if it was not indexed before. After indexing is done it delegate the search request to the wrapped
-     * {@link MessageSearchIndex}. Be aware that concurrent search requests are blocked on the same "not-yet-indexed" mailbox till it the index process was 
-     * complete
-     * 
-     */
-    @Override
-    public Stream<MessageUid> search(MailboxSession session, Mailbox mailbox, SearchQuery searchQuery) throws MailboxException {
-        Preconditions.checkArgument(session != null, "'session' is mandatory");
-        MailboxId id = mailbox.getMailboxId();
-        
-        Object done = indexed.get(id);
-        if (done == null) {
-            done = new Object();
-            Object oldDone = indexed.putIfAbsent(id, done);
-            if (oldDone != null) {
-                done = oldDone;
-            }
-            synchronized (done) {
-                Iterator<MailboxMessage> messages = factory.getMessageMapper(session).findInMailbox(mailbox, MessageRange.all(), FetchType.Full, UNLIMITED);
-                while (messages.hasNext()) {
-                    final MailboxMessage message = messages.next();
-                    try {
-                        add(session, mailbox, message).block();
-                    } catch (Exception e) {
-                        LOGGER.error("Unable to index message {} in mailbox {}", message.getUid(), mailbox.getName(), e);
-                    }
-                }
-            }
-        }
-       
-        return index.search(session, mailbox, searchQuery);
-    }
-
-    @Override
-    public Mono<Void> update(MailboxSession session, Mailbox mailbox, List<UpdatedFlags> updatedFlagsList) {
-        return index.update(session, mailbox, updatedFlagsList);
-    }
-    
-
-    @Override
-    public Flux<MessageId> search(MailboxSession session, Collection<MailboxId> mailboxIds, SearchQuery searchQuery, long limit) throws MailboxException {
-        throw new UnsupportedSearchException();
-    }
-}
+/****************************************************************
```

Comment:
There likely is a linebreak change here, be sure to commit it separately to ease review (sorry)